docs/devel/kubemark-guide.md

```diff
@@ -0,0 +1,139 @@
+     width="25" height="25">
+<img src="http://kubernetes.io/img/warning.png" alt="WARNING"
+     width="25" height="25">
+
+<h2>PLEASE NOTE: This document applies to the HEAD of the source tree</h2>
+
+If you are using a released version of Kubernetes, you should
+refer to the docs that go with that version.
+
+<strong>
+The latest 1.0.x release of this document can be found
+[here](http://releases.k8s.io/release-1.0/docs/devel/kubemark-guide.md).
+
+Documentation for other releases can be found at
+[releases.k8s.io](http://releases.k8s.io).
+</strong>
+--
+
+<!-- END STRIP_FOR_RELEASE -->
+
+<!-- END MUNGE: UNVERSIONED_WARNING -->
+
+# Kubemark User Guide
+
+## Introduction
+
+Kubemark is a performance testing tool which allows users to run experiments on simulated clusters. It’s main use case is scalability testing, as simulated clusters can be
+much bigger than the real ones. On the other hand it’s not designed for correctness testing, but nethertheless it can help with it, by exposing some problems on master
+components (API server, controller manager or scheduler) that appear only on bigger clusters (e.g. small memory leaks).
+
+This guide will help you understand what the Kubemark is, what it is not, and how to use it.
+
+## Architecture
+
+On a very high level Kubemark cluster consists of two parts: real master components and a set of “Hollow” Nodes. The prefix “Hollow” means an implementation/instantiation of a
+component with all “moving” parts mocked out. The best example is HollowKubelet, which pretends to be an ordinary Kubelet, but does not start anything, nor mount any volumes -
+it just lies it does. More detailed design and implementation details are at the end of this document.
+
+Currently master components run on a dedicated machine(s), and HollowNodes run on an ‘external’ Kubernetes cluster. This design has a slight advantage, over running master
+components on external cluster, of completely isolating master resources from everything else.
+
+## Requirements
+
+To run Kubemark you need a Kubernetes cluster for running all your HollowNodes and a dedicated machine for a master. Master machine has to be directly routable from
+HollowNodes. You also need an access to some Docker repository.
+
+Currently scripts are written to be easily usable by GCE, but it should be relatively straightforward to port them to different providers or bare metal.
+
+## Common use cases and helper scripts
+
+Common workflow for Kubemark is:
+- starting a Kubemark cluster (on GCE)
+- running e2e tests on Kubemark cluster
+- monitoring test execution and debugging problems
+- turning down Kubemark cluster
+
+In the text there will be comments helpful for anyone who’ll want to port Kubemark to different providers.
+
+### Starting a Kubemark cluster
+
+To start a Kubemark cluster on GCE you need to create an external cluster (it can be GCE or GKE cluster) by yourself and run `test/kubemark/start-kubemark.sh` script. This
```

Comment:
Why it needs to be GCE or GKE cluster? What needs to be done to use e.g. cluster running on bare-metal?